platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java

```diff
@@ -394,9 +394,9 @@ public Response retrieveMetadata() throws WSSecurityException, CertificateEncodi
     X509Certificate encryptionCert =
         findCertificate(systemCrypto.getEncryptionAlias(), systemCrypto.getEncryptionCrypto());
 
-    String hostname = SystemBaseUrl.getHost();
-    String port = SystemBaseUrl.getPort();
-    String rootContext = SystemBaseUrl.getRootContext();
+    String hostname = SystemBaseUrl.INTERNAL.getHost();
```

Comment:
❓General question for @stustison. Looks like we're exposing the IdP using the internal URL, and the client code is also using the internal URL to access it. Will that cause problems in an HA configuration?

Reply:
in HA, we expected the IdP to always respond to the external URL which is why that hostname had to be changed to match the external visible URL and not the internal one.

Reply:
I think it would cause issues in any configuration, depends on what is getting handed back to the user's browser during the IdP redirect. I can't remember where that url comes from though.

Reply:
We don't expose enough url configurations to allow IdP to work with a reverse proxy today. I think that would be additional follow on work to this task.
For now, I've left all of these as internal.

Reply:
fyi i _think_ this is determines the url that the user gets redirected to: https://github.com/codice/ddf/blob/master/platform/security/idp/security-idp-client/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L33

Reply:
Yeah this doesn't seem quite right to me for all of the IdP configuration. It looks like all of the IdP stuff is pointing towards the internal properties, but all of it gets proxied through the browser, so will go through the external interfaces. Seems like this is wrong to me. Not just here, but all the stuff below for logging in and logging out.

Reply:
I think the intent was to wait until we make sure it's even possible to configure our IdP to work behind a proxy. My understanding is that that scenario does not work currently.